src/test/java/seedu/address/model/position/UniquePositionListTest.java

```diff
@@ -0,0 +1,173 @@
+package seedu.address.model.position;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_TITLE_ADMIN_ASSISTANT;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalPositions.ADMIN_ASSISTANT;
+import static seedu.address.testutil.TypicalPositions.BOOKKEEPER;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.position.exceptions.DuplicatePositionException;
+import seedu.address.model.position.exceptions.PositionNotFoundException;
+import seedu.address.testutil.PositionBuilder;
+
+public class UniquePositionListTest {
+
+    private final UniquePositionList uniquePositionList = new UniquePositionList();
+
+    @Test
+    public void contains_nullPosition_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniquePositionList.contains(null));
+    }
+
+    @Test
+    public void contains_positionNotInList_returnsFalse() {
+        assertFalse(uniquePositionList.contains(ADMIN_ASSISTANT));
+    }
+
+    @Test
+    public void contains_positionInList_returnsTrue() {
+        uniquePositionList.add(ADMIN_ASSISTANT);
+        assertTrue(uniquePositionList.contains(ADMIN_ASSISTANT));
+    }
+
+    @Test
+    public void contains_positionWithSameIdentityFieldsInList_returnsTrue() {
+        uniquePositionList.add(ADMIN_ASSISTANT);
+        Position editedAdminAssistant = new PositionBuilder(ADMIN_ASSISTANT)
+                .withTitle(VALID_TITLE_ADMIN_ASSISTANT)
+                .build();
+        assertTrue(uniquePositionList.contains(editedAdminAssistant));
+    }
+
+    @Test
+    public void add_nullPosition_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniquePositionList.add(null));
+    }
+
+    @Test
+    public void add_duplicatePosition_throwsDuplicatePositionException() {
+        uniquePositionList.add(ADMIN_ASSISTANT);
+        assertThrows(DuplicatePositionException.class, () -> uniquePositionList.add(ADMIN_ASSISTANT));
+    }
+
+    @Test
+    public void setPosition_nullTargetPerson_throwsNullPointerException() {
```

Comment:
perhaps "person" should be "position" here?

Reply:
Changes made accoordinly.